src/Grid/grid.jl

```diff
@@ -1450,3 +1605,76 @@ struct SurfaceOrientationInfo
     #flipped::Bool
     #shift_index::Int
 end
+# This one is the same as `SurfaceOrientationInfo` but used in interfaces. It is calculated for the other facet relative to current facet
+struct InterfaceOrientationInfo
+    flipped::Bool
+    transformation::Union{Nothing, AbstractArray}
+end
+
+function transform_coordinates(src::Vector{Vector{Float64}}, dst::Vector{Vector{Float64}})
```

Comment:
Any reason why this should be vector of vectors instead of a matrix or tensor?

Reply:
Maybe change it to `Vec{3, Vec{2, Float64}}`? as it's always a 3 element vector of 2D points. I assume this won't work with `reverse!` because it's mutating but I'll see how things play out.